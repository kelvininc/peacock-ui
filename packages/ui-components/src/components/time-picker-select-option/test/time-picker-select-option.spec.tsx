import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { KvTimePickerSelectOption } from '../time-picker-select-option';
import { h } from '@stencil/core';

describe('KvTimePickerSelectOption (unit tests)', () => {
	let page: SpecPage;
	let component: KvTimePickerSelectOption;

	describe('when rendering with default props', () => {
		beforeEach(async () => {
			page = await newSpecPage({
				components: [KvTimePickerSelectOption],
				template: () => <kv-time-picker-select-option label="Option 1" value="option1"></kv-time-picker-select-option>
			});
			component = page.rootInstance;
		});

		describe('when the component loads', () => {
			it('should match the snapshot', () => {
				expect(page.root).toMatchSnapshot();
			});

			it('should set the label', () => {
				expect(component.label).toEqual('Option 1');
			});

			it('should set the option value', () => {
				expect(component.value).toEqual('option1');
			});

			it('should set the selected status', () => {
				expect(component.selected).toEqual(false);
			});
		});
	});
});
