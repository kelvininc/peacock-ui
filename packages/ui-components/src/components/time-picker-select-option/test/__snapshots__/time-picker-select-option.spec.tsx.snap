// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KvTimePickerSelectOption (unit tests) when rendering with default props when the component loads should match the snapshot 1`] = `
<kv-time-picker-select-option label="Option 1" value="option1">
  <mock:shadow-root>
    <div class="select-option">
      <div class="text-container">
        <div class="item-label">
          Option 1
        </div>
        <div class="item-description"></div>
      </div>
    </div>
  </mock:shadow-root>
</kv-time-picker-select-option>
`;
